File: Java/labs/lab05_Prototype/task_3_2/src/SalesTeam.java
import java.util.ArrayList;
import java.util.List;

public class SalesTeam {

    private final List<Manager> managers = new ArrayList<>();
    private final List<Salesperson> salespeople = new ArrayList<>();

    void addManager(Manager manager) {
        managers.add(manager);
    }

    void addSalesperson(Salesperson salesperson) {
        salespeople.add(salesperson);
    }

    public void payExpenses(int amount) {
        for (Manager manager : managers) {
            manager.payExpenses(amount);
        }

        for (Salesperson salesperson : salespeople) {
            salesperson.payExpenses(amount);
        }
    }

}

File: Cpp/lab09_Composite/task_3_2/src/sales_team.cpp
#include "sales_team.hpp"


SalesTeam::SalesTeam() {}

void SalesTeam::addManager(Manager* manager)
{
  this->managers.push_back(manager);
}

void SalesTeam::addSalesperson(Salesperson* salesperson)
{
  this->salespeople.push_back(salesperson);
}

void SalesTeam::payExpenses(int amount)
{
  for (auto& manager: this->managers)
  {
    manager->payExpenses(amount);
  }
  for (auto& salesperson : this->salespeople)
  {
    salesperson->payExpenses(amount);
  }
}
